src/coreclr/inc/clr_std/vector

```diff
@@ -1,462 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
-//
-// clr_std/vector
-//
-// Copy of some key Standard Template Library functionality
-//
-
-#ifdef _MSC_VER
-#pragma once
-#endif
-
-#ifdef USE_STL
-#include <vector>
-#else
-#ifndef __clr_std_vector_h__
-#define __clr_std_vector_h__
-
-// This is defined in the debugmacrosext.h header, but don't take a dependency on that.
-#ifndef INDEBUG
-#ifdef _DEBUG
-#define INDEBUG(x)          x
-#else
-#define INDEBUG(x)
-#endif
-#endif // !def INDEBUG
-
-namespace std
-{
-    template <class T>
-    class vector
```

Comment:
Is switching to STL definition of collection classes going to introduce STL exception handling? The runtime exception handling is not compatible with that.

Reply:
I think it might, but we can't switch to using the CRT headers without switching these headers as well due to some build issues.

I'll keep some of these headers (the ones that are used in the runtime and use components that would use C++ exception handling) and change the namespace to something like `clr_std` to avoid conflicting with the STL.

Reply:
After investigating further, the only place in the repo that used any of the STL facilities that would introduce exception handling is superpmi. The only usages within the rest of the CoreCLR codebase of the `clr_std` headers were compile-time type traits and `std::move`. Therefore, I believe we can remove all of these headers without issue (and be cognizant in the future around introducing STL usage that may introduce exceptions).